modules/TargetSafety.py

```diff
@@ -0,0 +1,305 @@
+        assay_component_endpoint_name | ACEA_ER_80hr
+        assay_component_desc          | ACEA_ER_80hr, is ...
+        biological_process_target     | cell proliferation
+        tissue                        | null
+        cell_format                   | cell line
+        cell_short_name               | T47D
+        assay_format_type             | cell-based
+        official_symbol               | ESR1
+        eventId                       | null
+
+    Ex. output record:
+     targetFromSourceId | ESR1
+    event              | cell proliferation
+    eventId            | null
+    biosample          | {null, null, T47D...
+    datasource         | ToxCast
+    url                | https://www.epa.g...
+    study              | {ACEA_ER_80hr, AC...
+    """
+
+    return spark.read.csv(toxcast, sep='\t', header=True).select(
+        F.trim(F.col('official_symbol')).alias('targetFromSourceId'),
+        F.col('biological_process_target').alias('event'),
+        'eventId',
+        F.struct(
+            F.col('tissue').alias('tissueLabel'),
+            F.lit(None).alias('tissueId'),
+            F.col('cell_short_name').alias('cellLabel'),
+            F.col('cell_format').alias('cellFormat'),
+            F.lit(None).alias('cellId'),
+        ).alias('biosample'),
+        F.lit('ToxCast').alias('datasource'),
+        F.lit('https://www.epa.gov/chemical-research/exploring-toxcast-data-downloadable-data').alias('url'),
+        F.struct(
+            F.col('assay_component_endpoint_name').alias('name'),
+            F.col('assay_component_desc').alias('description'),
+            F.col('assay_format_type').alias('type'),
+        ).alias('study'),
+    )
+
+
+def initialize_spark():
+    """Spins up a Spark session."""
+
+    # Initialize spark session
+    spark_mem_limit = detect_spark_memory_limit()
+    spark_conf = (
+        SparkConf()
+        .set('spark.driver.memory', f'{spark_mem_limit}g')
+        .set('spark.executor.memory', f'{spark_mem_limit}g')
+        .set('spark.driver.maxResultSize', '0')
+        .set('spark.debug.maxToStringFields', '2000')
+        .set('spark.sql.execution.arrow.maxRecordsPerBatch', '500000')
+    )
+    spark = SparkSession.builder.config(conf=spark_conf).master('local[*]').getOrCreate()
+    logging.info(f'Spark version: {spark.version}')
+
+    return spark
+
+def get_parser():
+    """Get parser object for script TargetSafety.py."""
+    parser = argparse.ArgumentParser(description=__doc__)
+
+    parser.add_argument(
+        '--toxcast',
+        help='Input table containing biological processes associated with relevant targets that have been observed in toxicity assays.',
+        type=str,
+        required=True,
+    )
+    parser.add_argument(
+        '--adverse_events',
+        help='Input TSV containing adverse events associated with targets that have been collected from relevant publications. Fetched from GitHub..',
+        type=str,
+        required=True,
```

Comment:
I think this is not good. If the github file is hardcoded, this argumentum isn't required. However, I would rather keep this argument mandatory and remove the hardcoded variable and add the path to the config file.